### PR TITLE
Markf/demographics

### PR DESCRIPTION
**Pull request by 2shortplanks**

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -1,5 +1,11 @@
 {{$NEXT}}
 
+- Added support for 'average_income' and 'estimated_population' in the
+  Insights location record
```
Comment (oschwald) [inline Changes]: We should do a minor version bump as opposed to the default patch bump as we are ostensibly following Semantic Versioning for this dist.

File: lib/GeoIP2/Record/Location.pm [hunk 2]
```diff
@@ -15,6 +15,18 @@ has accuracy_radius => (
...
 
+has average_income => (
+    is        => 'ro',
+    isa       => NonNegativeInt,
+    predicate => 'has_average_income',
+);
+
+has estimated_population => (
+    is        => 'ro',
+    isa       => NonNegativeInt,
```
Comment (oschwald) [inline lib/GeoIP2/Record/Location.pm]: Out of curiosity, do we ever return 0 for either of these?

File: lib/GeoIP2/Record/Location.pm [hunk 3]
```diff
@@ -78,6 +90,20 @@ IP address is likely to be.
...
+
+This returns a non-negative integer representing the average income in US
+dollars associated with the requested IP address.
+
+This attribute is only available from the Insights end point.
+
+=head2 $location_rec->estimated_population()
+
+Returns a non-negative integer representing the estimated population per square
+mile associated with the requested IP address.
```
Comment (oschwald) [inline lib/GeoIP2/Record/Location.pm]: kilometer, not mile, right?

File: t/lib/Test/GeoIP2.pm [hunk 4]
```diff
@@ -17,6 +17,7 @@ our @EXPORT_OK = qw(
 sub test_model_class {
     my $class = shift;
     my $raw   = shift;
+    my $uboat = shift;
```
Comment (oschwald) [inline t/lib/Test/GeoIP2.pm]: This could probably be named more clearly.

File: lib/GeoIP2/Record/Location.pm (hunk 2 above)
Comment (2shortplanks) [inline lib/GeoIP2/Record/Location.pm]: No

File: lib/GeoIP2/Record/Location.pm (hunk 3 above)
Comment (2shortplanks) [inline lib/GeoIP2/Record/Location.pm]: Good catch. That's documentation left over from before we converted to using kilometres

File: lib/GeoIP2/Record/Location.pm (hunk 2 above)
Comment (oschwald) [inline lib/GeoIP2/Record/Location.pm]: Should we say positive in the docs and use stricter type constraints?

Comment (2shortplanks) [inline lib/GeoIP2/Record/Location.pm]: One day our datasources could theoretically have valid `0` values (for example an area that's populated by people who don't work, or an area where people only work but no one actually lives there.)  Let's leave it as it is.

**Pull request merged**